hie-compat/src-ghc86/Compat/HieDebug.hs

```diff
@@ -129,9 +129,7 @@ validateScopes asts = M.foldrWithKey (\k a b -> valid k a ++ b) [] refMap
     valid (Right n) refs = concatMap inScope refs
       where
         mapRef = foldMap getScopeFromContext . identInfo . snd
-        scopes = case foldMap mapRef refs of
-          Just xs -> xs
-          Nothing -> []
+        scopes = Data.Maybe.fromMaybe [] (foldMap mapRef refs)
```

Comment:
FWIW I personally disagree with this hlint rule pretty strongly. If anything I tend to replace `fromMaybe` and friends with explicit pattern matches unless there's something higher-order going on!

Reply:
Main issue I think that needs to be addressed is the difference between what is configured in `.htlint.yaml` and the codebase. I myself am trying to be not opinionated in this PR. But I'm also not sure regarding defaults and capabilities of different hlint versions - and how hint levels are dealt with by different tools (hls, the github action in this repository and together with apply-refact). Currently I'm still looking into it.